src/Fixer/ClassNotation/ClassAttributesSeparationFixer.php

```diff
@@ -36,6 +36,13 @@
  * Make sure there is one blank line above and below class elements.
  *
  * The exception is when an element is the first or last item in a 'classy'.
+ *
+ * @phpstan-type ClassToken array{
```

Comment:
I would suggest to externalize this commit to another PR and remove it from this PR, to move faster.

In general, I'm against phpstan-based annotations like this, it's workaround of issue of user assoc arrays instead of DTO

Reply:
Can't we just leave it here just to have less PHPStan errors, but add `@todo introduce proper DTO`? I can remove it from this PR, but it will make baseline bigger.

Reply:
so...
> I would suggest to externalize this commit to another PR and remove it from this PR, to move faster.

------------

> Can't we just leave it here just to have less PHPStan errors

I see no value of removing phpstan errors for sake of removing phpstan errors. One either improve the codebase, and thanks to that phpstan reports one issue less, or not. Here, you are not improving the codebase, but making phpstan happy by a workaround of "introducing", but not introducing, the type.

And then we have things like:
```php
     * @param ClassToken $class
     */
    private function fixSpaceAboveClassElement(Tokens $tokens, array $class, int $elementIndex): void
```

which are contradicting about argument type [phpdoc says param is instance of given class, while it is not].

we want to solve issue that phpstan reports? it's great, but let's do it properly instead of bypassing the typing system

Reply:
I changed it to ~~redundant~~ reused array shape, without PHPStan alias, which could be confusing. So now we know what's inside the array, but we don't "pretend" it's an object or something - it's explicitly array, but with known keys/value types.